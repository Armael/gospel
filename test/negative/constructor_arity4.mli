type t = L of t * t | E

val f : t -> int
(*@ r = f a
    ensures
      match a with
      | L ((x,y)) -> x=y
*)
