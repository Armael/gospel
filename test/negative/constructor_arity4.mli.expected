
*******************************
********** Parsed file ********
*******************************
type t =
  | L of t * t 
  | E 
val f : t -> int[@@gospel
                  {| r = f a
    ensures
      match a with
      | L ((x,y)) -> x=y
|}]

*******************************
****** GOSPEL translation *****
*******************************
(*@ open Gospelstdlib *)

type t =
| L of t * t 
| E 
  

val f : t -> int
(*@ r = f a
    ensures ...
     *)
File "constructor_arity4.mli", line 7, characters 8-17:
Error: Function L expects 2 arguments as opposed to 1
